### ISSUE TITLE
Reproducible deadlock on Eclipse startup

### ISSUE THREAD

**Issue by peterkir** [bndtools/bnd#2888]
OS:  win7
Eclipse 2018.12
bndtools 4.0.0.201805111802
Git-Descriptor: 4.0.0.REL
Git-SHA: 64633ee367adab9d0f3dec865d62f943f0808b09
Java Oracle 1.8.0_144

After launching the Eclipse Workspace I always have a frozen Eclipse. 
JMC is showing the following blocked threads. 

Workspace contains a mixture of the following repo types.

- aQute.lib.deployer.FileRepo (old OSGi build repo)
- aQute.bnd.deployer.repository.LocalIndexedRepo (empty)
- multiple aQute.bnd.repository.p2.provider.P2Repository (Eclipse Platform, Simu, ...)
- aQute.bnd.repository.osgi.OSGiRepository

Baselining is active.

```
Stack Trace
main [1] (BLOCKED)
   bndtools.central.Central.getWorkspace line: 213 
   bndtools.central.Central.getWorkspaceBuildFile line: 168 
   bndtools.OpenExtConfigsContributionItem.getContributionItems line: 37 
   org.eclipse.ui.actions.CompoundContributionItem.getContributionItemsToFill line: 83 
   org.eclipse.ui.actions.CompoundContributionItem.fill line: 57 
   org.eclipse.ui.internal.menus.DynamicMenuContributionItem.fill line: 150 
   org.eclipse.jface.action.MenuManager.doItemFill line: 731 
   org.eclipse.jface.action.MenuManager.update line: 813 
   org.eclipse.jface.action.MenuManager.update line: 672 
   org.eclipse.e4.ui.workbench.renderers.swt.MenuManagerRenderer.scheduleManagerUpdate line: 1222 
   org.eclipse.e4.ui.workbench.renderers.swt.MenuManagerRenderer.subscribeUIElementTopicAllRenderedVisibility line: 203 
   sun.reflect.GeneratedMethodAccessor16.invoke line: not available 
   sun.reflect.DelegatingMethodAccessorImpl.invoke line: not available 
   java.lang.reflect.Method.invoke line: not available 
   org.eclipse.e4.core.internal.di.MethodRequestor.execute line: 58 
   org.eclipse.e4.ui.internal.di.UIEventObjectSupplier$UIEventHandler$1.run line: 67 
   org.eclipse.swt.widgets.Synchronizer.syncExec line: 236 
   org.eclipse.ui.internal.UISynchronizer.syncExec line: 147 
   org.eclipse.swt.widgets.Display.syncExec line: 4580 
   org.eclipse.e4.ui.internal.workbench.swt.E4Application$1.syncExec line: 219 
   org.eclipse.e4.ui.internal.di.UIEventObjectSupplier$UIEventHandler.handleEvent line: 64 
   org.eclipse.equinox.internal.event.EventHandlerWrapper.handleEvent line: 205 
   org.eclipse.equinox.internal.event.EventHandlerTracker.dispatchEvent line: 203 
   org.eclipse.equinox.internal.event.EventHandlerTracker.dispatchEvent line: 1 
   org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent line: 233 
   org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous line: 151 
   org.eclipse.equinox.internal.event.EventAdminImpl.dispatchEvent line: 132 
   org.eclipse.equinox.internal.event.EventAdminImpl.sendEvent line: 75 
   org.eclipse.equinox.internal.event.EventComponent.sendEvent line: 44 
   org.eclipse.e4.ui.services.internal.events.EventBroker.send line: 55 
   org.eclipse.e4.ui.internal.workbench.UIEventPublisher.notifyChanged line: 63 
   org.eclipse.emf.common.notify.impl.BasicNotifierImpl.eNotify line: 424 
   org.eclipse.e4.ui.model.application.ui.impl.UIElementImpl.setVisible line: 349 
   org.eclipse.e4.ui.workbench.renderers.swt.ContributionRecord.updateVisibility line: 113 
   org.eclipse.e4.ui.workbench.renderers.swt.MenuManagerRendererFilter.updateElementVisibility line: 194 
   org.eclipse.e4.ui.workbench.renderers.swt.MenuManagerRendererFilter.updateElementVisibility line: 205 
   org.eclipse.ui.internal.WorkbenchWindow.lambda$0 line: 714 
   org.eclipse.ui.internal.WorkbenchWindow$$Lambda$285/2105051301.run line: not available 
   org.eclipse.swt.widgets.RunnableLock.run line: 40 
   org.eclipse.swt.widgets.Synchronizer.runAsyncMessages line: 185 
   org.eclipse.swt.widgets.Display.runAsyncMessages line: 3919 
   org.eclipse.swt.widgets.Display.readAndDispatch line: 3550 
   org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run line: 1173 
   org.eclipse.core.databinding.observable.Realm.runWithDefault line: 339 
   org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run line: 1062 
   org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI line: 156 
   org.eclipse.ui.internal.Workbench.lambda$3 line: 636 
   org.eclipse.ui.internal.Workbench$$Lambda$85/1847905873.run line: not available 
   org.eclipse.core.databinding.observable.Realm.runWithDefault line: 339 
   org.eclipse.ui.internal.Workbench.createAndRunWorkbench line: 563 
   org.eclipse.ui.PlatformUI.createAndRunWorkbench line: 151 
   org.eclipse.ui.internal.ide.application.IDEApplication.start line: 155 
   org.eclipse.equinox.internal.app.EclipseAppHandle.run line: 203 
   org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication line: 137 
   org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start line: 107 
   org.eclipse.core.runtime.adaptor.EclipseStarter.run line: 400 
   org.eclipse.core.runtime.adaptor.EclipseStarter.run line: 255 
   sun.reflect.NativeMethodAccessorImpl.invoke0 line: not available [native method]
   sun.reflect.NativeMethodAccessorImpl.invoke line: not available 
   sun.reflect.DelegatingMethodAccessorImpl.invoke line: not available 
   java.lang.reflect.Method.invoke line: not available 
   org.eclipse.equinox.launcher.Main.invokeFramework line: 659 
   org.eclipse.equinox.launcher.Main.basicRun line: 595 
   org.eclipse.equinox.launcher.Main.run line: 1501 
```

```
Stack Trace
Worker-1: Updating workspace [30] (BLOCKED)
   bndtools.central.Central.getWorkspace line: 213 
   bndtools.central.Central.getProject line: 573 
   bndtools.central.Central.getProject line: 577 
   org.bndtools.builder.classpath.BndContainerInitializer$Updater.<init> line: 207 
   org.bndtools.builder.classpath.BndContainerInitializer.initialize line: 78 
   org.eclipse.jdt.internal.core.JavaModelManager.initializeContainer line: 3195 
   org.eclipse.jdt.internal.core.JavaModelManager.getClasspathContainer line: 2116 
   org.eclipse.jdt.core.JavaCore.getClasspathContainer line: 3661 
   org.eclipse.jdt.internal.core.JavaProject.resolveClasspath line: 3156 
   org.eclipse.jdt.internal.core.JavaProject.resolveClasspath line: 3320 
   org.eclipse.jdt.internal.core.ClasspathChange.generateDelta line: 243 
   org.eclipse.jdt.internal.core.DeltaProcessor.resourceChanged line: 2111 
   org.eclipse.jdt.internal.core.DeltaProcessingState.resourceChanged line: 477 
   org.eclipse.core.internal.events.NotificationManager$1.run line: 300 
   org.eclipse.core.runtime.SafeRunner.run line: 45 
   org.eclipse.core.internal.events.NotificationManager.notify line: 290 
   org.eclipse.core.internal.events.NotificationManager.broadcastChanges line: 153 
   org.eclipse.core.internal.resources.Workspace.broadcastPostChange line: 379 
   org.eclipse.core.internal.resources.Workspace.endOperation line: 1502 
   org.eclipse.core.internal.resources.Workspace.run line: 2308 
   org.eclipse.core.internal.events.NotificationManager$NotifyJob.run line: 44 
   org.eclipse.core.internal.jobs.Worker.run line: 63 

```

```
Stack Trace
Worker-4: Initialize CompilationUnit Manager [33] (BLOCKED)
   bndtools.central.Central.getWorkspace line: 213 
   bndtools.central.Central.getProject line: 573 
   bndtools.central.Central.getProject line: 577 
   org.bndtools.builder.classpath.BndContainerInitializer$Updater.<init> line: 207 
   org.bndtools.builder.classpath.BndContainerInitializer.initialize line: 78 
   org.eclipse.jdt.internal.core.JavaModelManager.initializeContainer line: 3195 
   org.eclipse.jdt.internal.core.JavaModelManager.getClasspathContainer line: 2116 
   org.eclipse.jdt.core.JavaCore.getClasspathContainer line: 3661 
   org.eclipse.jdt.internal.core.JavaProject.resolveClasspath line: 3156 
   org.eclipse.jdt.internal.core.JavaProject.resolveClasspath line: 3320 
   org.eclipse.jdt.internal.core.JavaProject.getResolvedClasspath line: 2397 
   org.eclipse.jdt.internal.core.JavaProject.buildStructure line: 478 
   org.eclipse.jdt.internal.core.Openable.generateInfos line: 268 
   org.eclipse.jdt.internal.core.Openable.openAncestors line: 530 
   org.eclipse.jdt.internal.core.Openable.generateInfos line: 250 
   org.eclipse.jdt.internal.core.Openable.openAncestors line: 530 
   org.eclipse.jdt.internal.core.Openable.generateInfos line: 250 
   org.eclipse.jdt.internal.core.JavaElement.openWhenClosed line: 596 
   org.eclipse.jdt.internal.core.JavaElement.getElementInfo line: 326 
   org.eclipse.jdt.internal.core.JavaElement.getElementInfo line: 312 
   org.eclipse.ajdt.core.javaelements.AJCompilationUnitManager.createCU line: 190 
   org.eclipse.ajdt.core.javaelements.AJCompilationUnitManager.initCompilationUnits line: 236 
   org.eclipse.ajdt.ui.AspectJUIPlugin$1.run line: 318 
   org.eclipse.core.internal.jobs.Worker.run line: 63 
```

**Comment by bjhargrave** [comment]
Can you please try the latest Bndtools 4.2 which is our current development? 

Also, the threads above only show who is blocked but some other thread must own the monitor they are blocked on. So the full thread dump information is needed.

**Comment by bjhargrave** [comment]
There is not enough information here to debug. Can you please provide a small github repo which demonstrates the problem? Then we can investigate with Bndtools 4.2 snapshot which has a _lot_ of improvements over Bndtools 4.0 in this area . Thanks.

**Comment by bjhargrave** [comment]
Please open a new issue after testing with Bndtools 4.2. Thanks.
